app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -841,6 +835,14 @@ class BrowserTabFragment : Fragment(), FindListener, CoroutineScope, DaxDialogLi
             }
 
             it.setDownloadListener { url, _, contentDisposition, mimeType, _ ->
+                //handling target=_blank if link points to a file
+                if (viewModel.url == null) {     //in an empty tab
```

Comment:
Could we move this logic to the `ViewModel` and add a test for it?

You'd need to:
- Add a method in the `ViewModel` that handles when `viewmodel.url` is null and calls `viewModel.closeCurrentTab()`
- Create a new `Command` that will be returned from the new method in the `ViewModel`, something like `Command.FileDownload`
- Process that new Command in the `BrowserTabFragment` that calls `destroyWebView()`. 
- And finally add a test for all this logic

Reply:
Thanks for the detailed review @malmstein . I will try to implement this

Reply:
Great feedback @malmstein. As the approach here matches the code around it and potentially fixes a frustrating user bug, I'm happy to accept the PR once we have a solid approach even without this feedback in. The PR will add value, even if it doesn't quite match our ideal internal style.